File: 백준/Silver/1931. 회의실 배정/회의실 배정.py
# ## 백준 1931번 문제 - 그리디 알고리즘

n = int(input())
data = []
count = 0

for i in range(n):
    e1, e2 = map(int, input().split())
    data.append([e1,e2])

data.sort(key = lambda x: (x[1], x[0]))
prev_end = 0

for start, end in data:
    if start >= prev_end:
        count+=1
        prev_end = end

print(count)

File: 백준/Gold/1931. 회의실 배정/회의실 배정.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.List;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());

        List<int[]> data = new ArrayList<>();

        for (int i = 0; i < n; i++) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            int e1 = Integer.parseInt(st.nextToken());
            int e2 = Integer.parseInt(st.nextToken());
            data.add(new int[]{e1, e2});
        }

        data.sort((a, b) -> {
            if (a[1] == b[1]) return Integer.compare(a[0], b[0]);
            return Integer.compare(a[1], b[1]);
        });

        int count = 0;
        int prevEnd = 0;

        for (int[] meeting : data) {
            int start = meeting[0];
            int end = meeting[1];

            if (start >= prevEnd) {
                count++;
                prevEnd = end;
            }
        }

        System.out.println(count);
    }
}
